install/kubernetes/cilium/templates/cilium-configmap.yaml

```diff
@@ -425,10 +425,18 @@ data:
 {{- if .Values.encryption.enabled }}
   {{- if eq .Values.encryption.type "ipsec" }}
   enable-ipsec: {{ .Values.encryption.enabled | quote }}
+
+    {{- if and (hasKey .Values.encryption "mountPath") (hasKey .Values.encryption "keyFile") }}
```

Comment:
I apologize that this is super late in the review cycle. But I just realized this now while reading https://github.com/cilium/cilium/pull/15846#issuecomment-825793752

The way this is set up, `encryption.{mountPath,keyFile,interface}` etc always takes precedence over  `encryption.ipsec.{mountPath,keyFile,interface}`. However, because we provide default values for the former, I think it's not possible to actually override them.

Example on this branch:
```console
$ helm template cilium ./cilium --namespace kube-system --set encryption.enabled=true --set encryption.ipsec.keyFile=/this/value/is/ignored | grep ipsec-key-file
  ipsec-key-file: /etc/ipsec/keys
```

Reply:
Nice catch! But even if we swap the order of precedence, since also `encryption.ipsec.{mountPath,keyFile,interface}` have default values, we'd always override any value for `encryption.{mountPath,keyFile,interface}` with the default ones of `encryption.ipsec.{mountPath,keyFile,interface}` :thinking: 

I think we should both:
* give precedence to `encryption.ipsec.{mountPath,keyFile,interface}`
* remove the default values for `encryption.ipsec.{mountPath,keyFile,interface}`

In this way:
* if set, the `encryption.ipsec.*` options take precedence
* otherwise we fallback to the old `encryption.*` (either the user specified values or the default ones)

what do you think?

Reply:
Yup, I think your proposal makes sense. Once we deprecate the old values, we could even re-introduce the default values for the new ones!